### ISSUE TITLE
SImpleBodyParser doesn't handle improperly formatted JSON

### ISSUE THREAD

**Issue by 4no0p** [firebase/firebase-tools#1273]
### Environment info

**firebase-tools:** 6.8.0

**Platform:** macOS

### Test case

- Run a HTTPS cloud function locally.
- Make a POST request to the cloud function with an invalid JSON in body.
- It times out after a while.

### Steps to reproduce

Run `firebase functions:shell` to emulate the HTTPS cloud function locally.
```
> firebase functions:shell

i  functions: Preparing to emulate functions.
✔  functions: api
firebase > info: Application ready to serve requests.
info: Environment: development
```

Make a HTTP POST request and pass an invalid JSON in the body.
```
{
    "someKey": {{invalidJSON}}
}
```

The cloud function fails with the following error message.
```
error: { SyntaxError: Unexpected token { in JSON at position 16
    at JSON.parse (<anonymous>)
    at parse (xxx/node_modules/firebase-tools/node_modules/body-parser/lib/types/json.js:89:19)
    at xxx/node_modules/firebase-tools/node_modules/body-parser/lib/read.js:121:18
    at invokeCallback (xxx/node_modules/firebase-tools/node_modules/raw-body/index.js:224:16)
    at done (xxx/node_modules/firebase-tools/node_modules/raw-body/index.js:213:7)
    at IncomingMessage.onEnd (xxx/node_modules/firebase-tools/node_modules/raw-body/index.js:273:7)
    at emitNone (events.js:106:13)
    at IncomingMessage.emit (events.js:208:7)
    at endReadableNT (_stream_readable.js:1064:12)
    at _combinedTickCallback (internal/process/next_tick.js:139:11)
    at process._tickCallback (internal/process/next_tick.js:181:9)
  expose: true,
  statusCode: 400,
  status: 400,
  body: '{\n    "someKey": {{invalidJSON}}\n}',
  type: 'entity.parse.failed' }
info: Execution took 539979 ms, finished with status: 'timeout'
```

###  Expected behavior

Return `400 Bad Request` or let the cloud function manage the bad request.

### Actual behavior

The HTTP request times out after a while.

**Comment by abeisgoat** [comment]
This is an issue with `SimpleBodyParser`, we should move to the full `bodyparser` package which should provide more consistent behavior.

**Comment by abeisgoat** [comment]
Will be resolved with #1277

**Comment by abeisgoat** [comment]
This bug is actually out of date (for 6.8.0), but a similar issue exists in #1277, so I'm closing this in favor of tracking there.
